From 20d52573c607ea4e77901167c83e9775ea640777 Mon Sep 17 00:00:00 2001
From: Laurent Pinchart <laurent.pinchart+renesas@ideasonboard.com>
Date: Sat, 8 Jun 2013 11:24:05 +0200
Subject: [PATCH 1032/1083] ARM: shmobile: lager: Add VSP1 support

Only the VSP1 DU0 is currently supported.

Signed-off-by: Laurent Pinchart <laurent.pinchart+renesas@ideasonboard.com>
(cherry picked from commit c92ff9101f0c4cd60afcb5e443c17f6d942bef1f)

Conflicts:
	arch/arm/mach-shmobile/board-lager.c

Signed-off-by: Simon Horman <horms+renesas@verge.net.au>
(cherry picked from commit 5d321bc7bc58800cdefff3d46b0aa0050a18c47a)

Conflicts:
	arch/arm/mach-shmobile/board-lager.c

Signed-off-by: Ryo Kataoka <ryo.kataoka.wt@renesas.com>
---
 arch/arm/mach-shmobile/board-lager.c | 32 ++++++++++++++++++++++++++++++++
 1 file changed, 32 insertions(+)

diff --git a/arch/arm/mach-shmobile/board-lager.c b/arch/arm/mach-shmobile/board-lager.c
index 3032385..4155550 100644
--- a/arch/arm/mach-shmobile/board-lager.c
+++ b/arch/arm/mach-shmobile/board-lager.c
@@ -29,6 +29,7 @@
 #include <linux/pinctrl/machine.h>
 #include <linux/platform_data/gpio-rcar.h>
 #include <linux/platform_data/rcar-du.h>
+#include <linux/platform_data/vsp1.h>
 #include <linux/platform_device.h>
 #include <linux/i2c.h>
 #include <linux/mtd/mtd.h>
@@ -181,6 +182,35 @@ static __initdata struct gpio_keys_platform_data lager_keys_pdata = {
 	.nbuttons	= ARRAY_SIZE(gpio_buttons),
 };
 
+/* VSP1 */
+static struct vsp1_platform_data lager_vspr_pdata = {
+	.features = 0,
+	.rpf_count = 5,
+	.uds_count = 1,
+	.wpf_count = 4,
+};
+
+static struct vsp1_platform_data lager_vsps_pdata = {
+	.features = 0,
+	.rpf_count = 5,
+	.uds_count = 3,
+	.wpf_count = 4,
+};
+
+static struct vsp1_platform_data lager_vspd0_pdata = {
+	.features = VSP1_HAS_LIF,
+	.rpf_count = 4,
+	.uds_count = 1,
+	.wpf_count = 4,
+};
+
+static struct vsp1_platform_data lager_vspd1_pdata = {
+	.features = VSP1_HAS_LIF,
+	.rpf_count = 4,
+	.uds_count = 1,
+	.wpf_count = 4,
+};
+
 static const struct pinctrl_map lager_pinctrl_map[] = {
 	/* DU (CN10: ARGB0, CN13: LVDS) */
 	PIN_MAP_MUX_GROUP_DEFAULT("rcar-du-r8a7790", "pfc-r8a7790",
@@ -466,6 +496,8 @@ static void __init lager_add_standard_devices(void)
 				      &lager_keys_pdata,
 				      sizeof(lager_keys_pdata));
 
+	r8a7790_add_vsp1_device(&lager_vspd0_pdata, 2);
+
 	/* QSPI flash memory */
 	spi_register_board_info(spi_info, ARRAY_SIZE(spi_info));
 
-- 
1.8.3.2

